controllers/monitoring_alert.go

```diff
@@ -303,14 +319,135 @@ func (r *ManagedMCGReconciler) reconcileAlertmanager() error {
 	return nil
 }
 
+// AlertRelabelConfigSecret will have configuration for relabeling the alerts that are firing.
+// It will add namespace label to firing alerts before they are sent to the alertmanager.
+func (r *ManagedMCGReconciler) reconcileAlertRelabelConfigSecret() error {
+	r.Log.Info("Reconciling alertRelabelConfigSecret")
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.alertRelabelConfigSecret, func() error {
+		if err := r.own(r.alertRelabelConfigSecret); err != nil {
+			return err
+		}
+		alertRelabelConfig := []struct {
+			TargetLabel string `yaml:"targetLabel,omitempty"`
+			Replacement string `yaml:"replacement,omitempty"`
+		}{{
+			TargetLabel: "namespace",
+			Replacement: r.namespace,
+		}}
+		config, err := yaml.Marshal(alertRelabelConfig)
+		if err != nil {
+			return fmt.Errorf("unable to encode alert relabel conifg: %w", err)
+		}
+		r.alertRelabelConfigSecret.Data = map[string][]byte{
+			alertRelabelConfigSecretKey: config,
+		}
+
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("unable to create/update AlertRelabelConfigSecret: %w", err)
+	}
+
+	return nil
+}
+
+func (r *ManagedMCGReconciler) reconcileKubeRBACConfigMap() error {
+	r.Log.Info("Reconciling kubeRBACConfigMap")
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.kubeRBACConfigMap, func() error {
+		if err := r.own(r.kubeRBACConfigMap); err != nil {
+			return err
+		}
+		r.kubeRBACConfigMap.Data = templates.KubeRBACProxyConfigMap.DeepCopy().Data
+
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("unable to create kubeRBACConfig config map: %w", err)
+	}
+
+	return nil
+}
+
+// reconcilePrometheusService function wait for prometheus Service
+// to start and sets appropriate annotation for 'service-ca' controller.
+func (r *ManagedMCGReconciler) reconcilePrometheusService() error {
+	r.Log.Info("Reconciling PrometheusService")
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.prometheusService, func() error {
+		if err := r.own(r.prometheusService); err != nil {
+			return err
+		}
+
+		r.prometheusService.Spec.Ports = []corev1.ServicePort{
+			{
+				Name:       "https",
+				Protocol:   corev1.ProtocolTCP,
+				Port:       int32(templates.KubeRBACProxyPortNumber),
+				TargetPort: intstr.FromString("https"),
+			},
+		}
+		r.prometheusService.Spec.Selector = map[string]string{
+			"app.kubernetes.io/name": r.prometheusService.Name,
+		}
+		utils.AddAnnotation(
+			r.prometheusService,
+			"service.beta.openshift.io/serving-cert-secret-name",
+			prometheusServiceSecretName,
+		)
+		utils.AddAnnotation(
+			r.prometheusService,
+			"service.alpha.openshift.io/serving-cert-secret-name",
+			prometheusServiceSecretName,
+		)
```

Comment:
Need to figure out why `serving-cert` is not being created even after adding these annotations.

<details>
<summary>Screenshot</summary>

![Screenshot from 2022-05-24 03-19-29](https://user-images.githubusercontent.com/33557095/169911481-40cbaa7d-f5f4-4f09-95dd-ab0700ace102.png)


</details>

Reply:
OpenShift service is supposed to add this secret, IIRC.